### PR TITLE
Depend on imageio that does not force numpy downgrade

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies = [
     'scipy>=1.9',
     'networkx>=2.8',
     'pillow>=9.1',
-    'imageio>=2.33',
+    'imageio>=2.35.1',
```
Comment (jakirkham) [inline pyproject.toml:33]:
```suggestion
    'imageio>=2.33,!=2.35.0',
```

File: requirements/default.txt [hunk 2]
```diff
@@ -4,7 +4,7 @@ numpy>=1.23
 scipy>=1.9
 networkx>=2.8
 pillow>=9.1
-imageio>=2.33
+imageio>=2.35.1
```
Comment (jakirkham) [inline requirements/default.txt:7]:
```suggestion
imageio>=2.33,!=2.35.0
```

**Review by jakirkham — COMMENTED** [review]
Thanks Stefan! 🙏

Looking at ImageIO's commit history, [the NumPy pin]( https://github.com/imageio/imageio/commit/6bc8c6cf9b5015eb4a514f8debcf92eca7162f1c#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R118-R122 ) was introduced in commit: https://github.com/imageio/imageio/commit/6bc8c6cf9b5015eb4a514f8debcf92eca7162f1c

So it looks like only `2.35.0` had this pin. Wonder if we can just exclude that version?

What do you think? 🙂

**Comment by stefanv** [comment]
That's better!

**Review by jakirkham — APPROVED** [review]

**Review by lagru — APPROVED** [review]
Looks good and agree that excluding 2.35.0 seems like the best approach.

**Comment by lagru** [comment]
Test failures are unrelated to this and are related to https://github.com/scikit-image/scikit-image/issues/7491. So merging. :)

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**